### PR TITLE
Style the Python code with Black

### PR DESCRIPTION
**Pull request by jeremycline**
This is a large commit, which is unfortunate for review, but its advantage is only one commit needs to be added to the revision ignore list for blame.

### CODE REVIEW

File: .git-blame-ignore-revs [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Migrate Python code to Black
```
Comment (dcantrell) [inline .git-blame-ignore-revs:1]: What does this file do?

Comment (jeremycline) [reply .git-blame-ignore-revs:1]: I left a note in the commit that formats the code on usage, you can pass it to git-blame so it ignores that particular commit. This way, if you're blame'ing around and it points at the reformat commit, you can look at that, get instructions on how to ignore it, and continue looking at blame info as if the commit never existed. Git documentation is https://git-scm.com/docs/git-blame#Documentation/git-blame.txt---ignore-revs-fileltfilegt

Comment (dcantrell) [reply .git-blame-ignore-revs:1]: OK, got it.  Nice.  Never used that functionality on git blame.

**Comment by jeremycline** [comment]
Yeah, I'm definitely not a fan of some of its style choices, but as much as I dislike them I dislike correcting people's style even more. Telling contributors to just add a pre-commit hook or similar is worth it.

**Pull request merged**